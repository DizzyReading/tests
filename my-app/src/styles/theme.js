import React from "react";
import { ThemeProvider } from "styled-components";
import PropTypes from "prop-types";
import theme from "../themes/default";
import GlobalStyles from "./globals";

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyles />
    {children}
  </ThemeProvider>
);

Theme.propTypes = {
  children: PropTypes.element.isRequired,
};

export default Theme;
