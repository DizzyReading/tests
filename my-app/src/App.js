/* eslint-disable no-unused-vars */
import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import SharedButton from "./components/Button/Button";
import Header from "./components/Header/Header";
import Headline from "./components/Headline/Headline";
import ListItem from "./components/ListItem/ListItem";
import { getPosts } from "./redux/posts/postsSlice";
import { Section } from "./styles/GlobalComps";

const tempArr = [
  {
    fName: "Jane",
    lName: "Doe",
    email: "janedoe@gmail.com",
    age: 23,
    onlineStatus: false,
  },
];

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    // setTimeout(() => {
    //   dispatch(getPosts());
    // }, 3000);
    // console.log("Actions:", fetchPosts(posts));
  }, []);

  const configButton = {
    buttonText: "Get Posts",
    emitEvent: getPosts,
  };

  const props = {
    header: "Posts",
    description: "Click the button to render posts!",
    tempArr,
  };
  return (
    <div className="App">
      <Header />
      <Section className="main">
        <Headline {...props} />
        <SharedButton {...configButton} />
      </Section>
    </div>
  );
}

export default App;
