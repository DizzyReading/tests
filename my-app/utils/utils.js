import React from "react";
import checkPropTypes from "check-prop-types";
import theme from "../src/themes/default";
import TestRenderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { configureStore } from "redux-mock-store";
import postsReducer from "../src/redux/posts/postsSlice";
import { createStoreHook } from "react-redux";
import { applyMiddleware, createStore } from "redux";
import ReduxThunk from "redux-thunk";

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );

  return propsErr;
};

export const renderWithTheme = (component) => {
  return TestRenderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

const middlewares = [ReduxThunk];

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(
    createStore
  );
  return createStoreWithMiddleware(postsReducer, initialState);
};
