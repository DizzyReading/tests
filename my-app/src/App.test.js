import React from "react";
import App from "./App";
import { shallow } from "enzyme";
import { findByTestAtrr, testStore } from "../utils/utils";
import theme from "./themes/default";
import toJson from "enzyme-to-json";

const setUp = (initialState = {}) => {
  let store = testStore(initialState);
  let wrapper = shallow(<App></App>);

  return wrapper;
};

describe("App Component", () => {
  let component;

  beforeEach(() => {
    const initialState = {
      posts: [
        {
          title: "Example Title 1",
          body: "Some Text",
        },
        {
          title: "Example Title 2",
          body: "Some Text",
        },
        {
          title: "Example Title 3",
          body: "Some Text",
        },
      ],
    };

    component = setUp(initialState);
  });

  it("Should render without errors", () => {
    const app = findByTestAtrr(component, "appComponent");
    expect(app.length).toBe(1);
    expect(toJson(app)).toMatchSnapshot();
  });
});
