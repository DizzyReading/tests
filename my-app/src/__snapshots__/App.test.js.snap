// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should render without errors 1`] = `
<div
  className="App"
  data-test="appComponent"
>
  <Header />
  <styled.section
    className="main"
  >
    <Headline
      description="Click the button to render posts!"
      header="Posts"
      tempArr={
        Array [
          Object {
            "age": 23,
            "email": "janedoe@gmail.com",
            "fName": "Jane",
            "lName": "Doe",
            "onlineStatus": false,
          },
        ]
      }
    />
    <SharedButton
      buttonText="Get Posts"
      emitEvent={[Function]}
    />
  </styled.section>
</div>
`;
